src/utilities/gpu_macro.cuh

```diff
@@ -0,0 +1,168 @@
+#define gpusolverSyevjInfo_t hipsolverSyevjInfo_t
+#define gpusolverDnCreateSyevjInfo hipsolverDnCreateSyevjInfo
+#define gpusolverDnDestroySyevjInfo hipsolverDnDestroySyevjInfo
+#define gpusolverDnZheevj_bufferSize hipsolverDnZheevj_bufferSize
+#define gpusolverDnZheevj hipsolverDnZheevj
+#define gpusolverDnZheevd_bufferSize hipsolverDnZheevd_bufferSize
+#define gpusolverDnZheevd hipsolverDnZheevd
+#define gpusolverDnDsyevj_bufferSize hipsolverDnDsyevj_bufferSize
+#define gpusolverDnDsyevj hipsolverDnDsyevj
+#define gpusolverDnZheevjBatched_bufferSize hipsolverDnZheevjBatched_bufferSize
+#define gpusolverDnZheevjBatched hipsolverDnZheevjBatched
+
+#else // CUDA for Nvidia card
+
+// memory manipulation
+#define gpuMalloc cudaMalloc
+#define gpuMallocManaged cudaMallocManaged
+#define gpuFree cudaFree
+#define gpuMemcpy cudaMemcpy
+#define gpuMemcpyFromSymbol cudaMemcpyFromSymbol
+#define gpuMemcpyToSymbol cudaMemcpyToSymbol
+#define gpuGetSymbolAddress cudaGetSymbolAddress
+#define gpuMemcpyHostToDevice cudaMemcpyHostToDevice
+#define gpuMemcpyDeviceToHost cudaMemcpyDeviceToHost
+#define gpuMemcpyHostToHost cudaMemcpyHostToHost
+#define gpuMemcpyDeviceToDevice cudaMemcpyDeviceToDevice
+#define gpuMemset cudaMemset
+
+// error handling
+#define gpuError_t cudaError_t 
+#define gpuSuccess cudaSuccess
+#define gpuGetErrorString cudaGetErrorString
+#define gpuGetLastError cudaGetLastError
+
+// device manipulation
+#define gpuSetDevice cudaSetDevice
+#define gpuGetDeviceCount cudaGetDeviceCount
+#define gpuDeviceProp cudaDeviceProp
+#define gpuGetDeviceProperties cudaGetDeviceProperties
+#define gpuDeviceCanAccessPeer cudaDeviceCanAccessPeer
+#define gpuDeviceEnablePeerAccess cudaDeviceEnablePeerAccess
+#define gpuDeviceSynchronize cudaDeviceSynchronize
+
+// stream
+#define gpuStream_t cudaStream_t
+#define gpuStreamCreate cudaStreamCreate
+#define gpuStreamDestroy cudaStreamDestroy
+
+// random numbers
+#define gpurandState curandState
+#define gpurand_normal_double curand_normal_double
+#define gpurand_normal curand_normal
+#define gpurand_init curand_init
+
+// blas
+#define gpublasHandle_t cublasHandle_t
+#define gpublasSgemv cublasSgemv
+#define gpublasSgemm cublasSgemm
+#define gpublasSdgmm cublasSdgmm
+#define gpublasDestroy cublasDestroy
+#define gpublasCreate cublasCreate
+#define GPUBLAS_SIDE_LEFT CUBLAS_SIDE_LEFT
+#define GPUBLAS_OP_N CUBLAS_OP_N
+
+// lapack
+#define gpuDoubleComplex cuDoubleComplex
+#define gpusolverDnHandle_t cusolverDnHandle_t
+#define gpusolverDnCreate cusolverDnCreate
+#define gpusolverDnDestroy cusolverDnDestroy
+#define gpusolverEigMode_t cusolverEigMode_t
+#define gpusolverFillMode_t cublasFillMode_t // why cublas?
+#define GPUSOLVER_EIG_MODE_NOVECTOR CUSOLVER_EIG_MODE_NOVECTOR
+#define GPUSOLVER_EIG_MODE_VECTOR CUSOLVER_EIG_MODE_VECTOR
+#define GPUSOLVER_FILL_MODE_LOWER CUBLAS_FILL_MODE_LOWER // why cublas?
+#define gpusolverSyevjInfo_t syevjInfo_t // why not cusolverSyevjInfo_t?
```

Comment:
why these questions in the comments? did you find an answer to them?

Reply:
perhaps they are just legacy names in CUDA.